PyInstaller/building/osx.py

```diff
@@ -17,15 +17,14 @@
 from PyInstaller.building.datastruct import TOC, Target, logger
 from PyInstaller.building.utils import (_check_path_overlap, _rmtree, add_suffix_to_extension, checkCache)
 from PyInstaller.compat import is_darwin
+from PyInstaller.config import CONF
```

Comment:
Ah, see this is one of the many reasons I hate the `CONF` object. The import genuinely does have to be inside `Bundle.__init__()` so that it is updated whenever we do nonsense like [this](https://github.com/pyinstaller/pyinstaller/blob/develop/tests/unit/test_hookutils.py#L113) to work around it. I believe this is the cause of the `KeyError`s you're getting.

Reply:
The only `KeyError` I see is in the Windows test suite, because I didn't remove the old test, which called into the building code directly with some assumptions. Unless I'm misreading the Mac test logs. I'm not sure why I'm getting 7 fails on Mac, it doesn't seem to give a reason.